Bundled Theme: Fix social links text decoration

SainathPoojary opened the pull request:
This pull request fixes an issue in the Twenty Nineteen theme where social icons in the wp-block-social-links block appear without underlines in the editor but are incorrectly displayed with underlines on the frontend. This PR resolves the issue by adding a text-decoration: none; rule for anchor tags in the frontend, ensuring consistent styling with the editor.

Trac ticket: [#61904](https://core.trac.wordpress.org/ticket/61904)

### Before:
<img width="1920" alt="image" src="https://github.com/user-attachments/assets/375a317a-36c1-40e7-a977-ba6423cc76c2">

### After:
<img width="1920" alt="image" src="https://github.com/user-attachments/assets/dddd9514-9605-4263-a88d-68b47c511ab8">

SainathPoojary committed:
Bundled Theme: Fix social links text decoration

Adds missing text decoration rule to the wp-block-social-links to ensure that social link anchors matches editor style.

rinkalpagdar commented:
Hello @SainathPoojary 
I have tested this PR change, it's working fine. 
Thanks

sabernhardt commented on src/wp-content/themes/twentynineteen/sass/blocks/_blocks.scss:
```diff
@@ -950,6 +950,11 @@
 		}
 	}
 
+	//! wp-block-social-links
```
```suggestion
	//! Social Icons
```

Use the block name for the comment.

sabernhardt commented on src/wp-content/themes/twentynineteen/sass/blocks/_blocks.scss:
```diff
@@ -950,6 +950,11 @@
 	}
 
+	//! wp-block-social-links
+	.wp-block-social-links .wp-block-social-link.wp-social-link .wp-block-social-link-anchor {
```
This creates a high specificity with six classes:
`.entry .entry-content .wp-block-social-links .wp-block-social-link.wp-social-link .wp-block-social-link-anchor`

An element selector (combined with `.entry .entry-content`) would be enough to override the underline in the post content:

```suggestion
	a:where(.wp-block-social-link-anchor) {
```

sabernhardt reviewed (changes requested):
Please update the comment, and _consider_ reducing the specificity.

If you could `run build` to update the stylesheets in this PR, that would help people testing it (and would help pass GitHub actions).

SainathPoojary committed:
Bundled Theme: Reduce specificity for social link styling and update comment

SainathPoojary commented:
Hey @sabernhardt,

I have implemented the requested changes and rebased the branch. After rebasing, I noticed that there’s some extra margin on the social icon button. Should I open a issue for it?

<img width="1920" alt="Screenshot 2025-02-21 at 2 28 42 PM" src="https://github.com/user-attachments/assets/678816ca-c58f-4970-929d-3c4ef47a361b" />

Additionally, while running npm install, I encountered a deprecation issue related to node-sass. If this needs to be addressed, I’d be happy to work on it. 

<img width="1323" alt="Screenshot 2025-02-21 at 2 52 38 PM" src="https://github.com/user-attachments/assets/712dd100-f774-45cc-9482-514ab0805688" />

sabernhardt commented:
The separate [Social Icons block margin issue](https://github.com/WordPress/gutenberg/issues/69098) should be fixed in Gutenberg 20.4.

I have had `node-sass` deprecation warnings too. In a previous installation I fixed it with the `rebuild` command, but I have not solved it in my new environment yet. I tried [replacing with `sass`](https://github.com/sabernhardt/wordpress-develop/pull/33/commits/63740a967afd5ec6e3ebb67cf85663900e206094) last year, but I did not like how much it changed in the compiled stylesheets just with the `package` changes. (Adding special rules might help if the switch is necessary.)

sabernhardt reviewed (approved):
Thanks!

These screenshots are all with the patch.

Front end, in the post content, in both language directions:

![post content, LTR](https://github.com/user-attachments/assets/1ea39ae6-aa78-418b-a255-e4210888f66b)

![post content, RTL](https://github.com/user-attachments/assets/efbf0a3c-9f37-4471-b361-011245378640)

The links were not underlined in the editor or in the widget context before the patch, but they still do not have the underline with the patch.

Post editor:

![post editor](https://github.com/user-attachments/assets/1051e5df-ea27-4041-b01d-8d6562211f1f)

Widget editor:

![widget in the editor](https://github.com/user-attachments/assets/cb05b4e1-8e4e-4937-8415-237669486879)

Widget on the front end:

![widget on front end](https://github.com/user-attachments/assets/22f754ad-e15b-4390-97a7-48ae5de3b2bd)
